diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/Generation.kt

```diff
@@ -56,20 +59,18 @@ private fun generateWarningNames() {
         .addComment(autoGenerationComment)
         .build()
 
-    kotlinFile.writeTo(File("diktat-rules/src/main/kotlin"))  // fixme: need to add it to pom
+    kotlinFile.writeTo(Paths.get("diktat-rules/src/main/kotlin"))  // fixme: need to add it to pom
 }
 
-@OptIn(ExperimentalPathApi::class)
 private fun validateYear() {
-    val files = File("diktat-rules/src/test/resources/test/paragraph2/header")
-    files
-        .listFiles()
-        ?.filterNot { it.name.contains("CopyrightDifferentYearTest.kt") }
-        ?.forEach { file ->
-            val tempFile = createTempFile().toFile()
-            tempFile.printWriter().use { writer ->
-                file.forEachLine { line ->
-                    writer.println(when {
+    val folder = Paths.get("diktat-rules/src/test/resources/test/paragraph2/header")
```

Comment:
This approach will work if _Maven_ is invoked from the project root, e. g.:

```console
$ mvn test
```
or

```console
$ mvn -pl diktat-rules test
```

Yet, a quite legitimate scenario (particularly, used by _IDEA_ when creating _Maven_ run configurations) is invoking a _Maven_ goal from the module directory, e. g.:

```console
cd diktat-rules/ && mvn test
```

In this latter case, _Maven_ execution will fail with:

```stacktrace
java.nio.file.NoSuchFileException: diktat-rules\src\test\resources\test\paragraph2\header
    at sun.nio.fs.WindowsException.translateToIOException (WindowsException.java:79)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:97)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:102)
    at sun.nio.fs.WindowsDirectoryStream.<init> (WindowsDirectoryStream.java:86)
    at sun.nio.fs.WindowsFileSystemProvider.newDirectoryStream (WindowsFileSystemProvider.java:518)
    at java.nio.file.Files.newDirectoryStream (Files.java:457)
    at java.nio.file.Files.list (Files.java:3451)
    at org.cqfn.diktat.ruleset.generation.GenerationKt.validateYear (Generation.kt:67)
    at org.cqfn.diktat.ruleset.generation.GenerationKt.main (Generation.kt:36)
    at org.cqfn.diktat.ruleset.generation.GenerationKt.main (Generation.kt)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:254)
    at java.lang.Thread.run (Thread.java:750)
```